packages/backend/src/dao/kv-dao.ts

```diff
@@ -1,64 +1,89 @@
-import { PolyratingsError } from "@polyratings/backend/utils/errors";
-import { DEFAULT_VALIDATOR_OPTIONS } from "@polyratings/backend/utils/const";
-import { validateOrReject } from "class-validator";
-import { transformAndValidate } from "@polyratings/backend/utils/transform-and-validate";
-import { plainToInstance } from "class-transformer";
-import { BulkKey, Internal } from "@polyratings/shared";
+import { BulkKey, BulkKeyMap } from "@backend/utils/const";
+import { z } from "zod";
+import { TRPCError } from "@trpc/server";
+import {
+    PendingRating,
+    pendingRatingValidator,
+    Professor,
+    professorValidator,
+    RatingReport,
+    ratingReportValidator,
+    TruncatedProfessor,
+    truncatedProfessorValidator,
+    User,
+    userValidator,
+} from "@backend/types/schema";
+import {
+    addRating,
+    pendingRatingToRating,
+    professorToTruncatedProfessor,
+    removeRating,
+} from "@backend/types/schemaHelpers";
+import { KvWrapper } from "./kv-wrapper";
 
 const KV_REQUESTS_PER_TRIGGER = 1000;
 
 export class KVDAO {
     constructor(
-        private polyratingsNamespace: KVNamespace,
-        private usersNamespace: KVNamespace,
-        private processingQueueNamespace: KVNamespace,
-        private professorApprovalQueueNamespace: KVNamespace,
-        private reportsNamespace: KVNamespace,
+        private polyratingsNamespace: KvWrapper,
+        private usersNamespace: KvWrapper,
+        private processingQueueNamespace: KvWrapper,
+        private professorApprovalQueueNamespace: KvWrapper,
+        private reportsNamespace: KvWrapper,
     ) {}
 
-    // HACK: class-validator/transformer cannot actually parse through the entire
-    // list of professors, so we just have to trust that it's actually valid/correct.
-    async getAllProfessors(): Promise<string> {
-        const professorList = await this.polyratingsNamespace.get("all");
+    async getAllProfessors() {
+        const professorList = await this.polyratingsNamespace.get(
+            z.array(truncatedProfessorValidator),
+            "all",
+        );
         if (!professorList) {
-            throw new PolyratingsError(404, "Could not find any professors.");
+            throw new TRPCError({
+                code: "INTERNAL_SERVER_ERROR",
+                message: "Could not find any professors.",
+            });
         }
```

Comment:
Why do we do this check only on `getAllProfessors`?

Reply:
Good catch! I meant to wrap it in a try catch since not finding all professors should throw a 500 instead of 404 since it is not something that should not just be "not found".